### DISCUSSION TITLE
Firebase auth/user data to Supabase auth/public.users tables

### DISCUSSION THREAD

**Discussion by JohnCarterGonzalez** [supabase/supabase#20450] [Questions]
I have a firebase project with about ~1200 profiles at the moment. I need to move the firebase auth and firestore data over to supabase. The firestore data contains in the users collection a nested collection of "user_types". I am wondering if these two transitions can be done harmlessly or if there are potential footguns I might need to look out for. When I migrate the auth table from firebase, is it a relatively straight forward process to create the fk relationship to the public.users table?

New to relational databases and supabase.

**Comment by GaryAustin1** [comment]
You can have one or more tables linked based on the user id.  You can either point them with a foreign key to auth.users, or you can have a main public table like users or profiles and point all other tables to that.  The 2nd solution allows you to join between those tables with the API, which can't be done if they join thru auth.users.

**Reply by JohnCarterGonzalez** [reply]
@GaryAustin1 the current thinking for the supabase side is to have three tables in the public schema. One for the users, one for user_type, and the other as an associative table, lets say "user_type_junction". If there is a nested collection, it seems relative straightforward to split the users collection into two tables through the [data-migration](https://supabase.com/docs/guides/resources/migrating-to-supabase/firestore-data) tool. Yet for three tables, seems a little more tricky. Maybe I'm over thinking it. Back to the auth, because Firebase users the same uuid for auth and (lets say) public collections, I'm assuming they would connect in the Supabase tables just fine? Once loaded in to their respective tables via the aforementioned data-migration tool? Thanks for help by the way!

**Reply by GaryAustin1** [reply]
I'm not that familiar with the tool.  If the user id is used to create the user in auth.users (I believe it is) then if you use that same user id in all your tables everything should be linked with a foreign key to a common point as I mentioned above.  Normally when you are using Supabase auth you would pre-populate tables with trigger functions as user's are added.  

Although nothing  is set in stone as you can do what you want with your tables.  The suggested user set up is discussed here: https://supabase.com/docs/guides/auth/managing-user-data .
It does not talk about the linking of other tables though besides the main profiles table.
